dart/lib/src/protocol/dsn.dart

```diff
@@ -14,34 +14,36 @@ class Dsn {
   final String publicKey;
 
   /// The Sentry.io secret key for the project.
-  final String secretKey;
+  final String? secretKey;
 
   /// The ID issued by Sentry.io to your project.
   ///
   /// Attached to the event payload.
   final String projectId;
 
   /// The DSN URI.
-  final Uri uri;
+  final Uri? uri;
 
-  String get postUri {
-    final port = uri.hasPort &&
-            ((uri.scheme == 'http' && uri.port != 80) ||
-                (uri.scheme == 'https' && uri.port != 443))
-        ? ':${uri.port}'
+  Uri get postUri {
```

Comment:
the signature change was required or just an improvement?

Reply:
This is because `http` now requires an Uri instead of String or Uri